relay-replays/src/serialization.rs

```diff
@@ -0,0 +1,149 @@
+use serde::de::Visitor;
+use serde::{Deserialize, Deserializer, Serialize};
+
+use crate::recording::{DocumentNode, DocumentTypeNode, ElementNode, NodeVariant, TextNode};
+
+impl<'de> Deserialize<'de> for NodeVariant {
```

Comment:
This implementation was copy-pasted and adapted from the expanded `derive(Deserialize)` of a simple, internally tagged enum.